docs/docs/schema/pipeline.json

```diff
@@ -91,6 +88,7 @@
             "enum": [
                 "input",
                 "extra",
+                "pattern",
```

Comment:
Why is there still extra? And why is there pattern and input-pattern?

Reply:
Otherwise there would be breaking changes

Reply:
I think that would be fine, user base is small. What do you think @disrupted @raminqaf ?

Reply:
Otherwise we keep carrying technical debt because decision in the beginning weren't ideal

Reply:
I am fine with it, but keep in mind we would have to do a major version bump for the next release, aka kpops 2.0

Reply:
our current plan was to make changes in a backwards compatible manner and then once those changes pile up make a breaking change for v2

Reply:
I consider KPOps still beta wrt to API and I think it is much more beneficial to release often and document breaking changes if needed. KPOps is not ages old :-)

Reply:
yeah, but we're not pre-release (0.x.x) anymore. Since we use semantic versioning we should follow its rules.

Reply:
I mean it is also fine to have v2 and quite soon v3

Reply:
yep, absolutely!

Reply:
Sounds good. Will carry out the breaking changes in this PR

Reply:
@philipp94831 I think I'm done with the changes to the code, what do you think?